mllib-local/src/main/scala/org/apache/spark/ml/linalg/Matrices.scala

```diff
@@ -999,7 +999,7 @@ object Matrices {
             val data = new ArrayBuffer[(Int, Int, Double)]()
             dnMat.foreachActive { (i, j, v) =>
               if (v != 0.0) {
-                data.append((i, j + startCol, v))
+                data.+=((i, j + startCol, v))
```

Comment:
> I think it's reasonable to fix this everywhere.
ok let me do that -- I was conservative in fixing only these instances that were in a loop.